app-modules/integration-ai/src/DataTransferObjects/DynamicContext.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Assist\IntegrationAI\DataTransferObjects;
+
+use Spatie\LaravelData\Data;
+use Illuminate\Database\Eloquent\Model;
+
+class DynamicContext extends Data
+{
+    public Model $record;
```

Comment:
I believe you could type hint this to `ProvidesDynamicContext` then in the `ProvidesDynamicContext` give it a docblock `@mixin` to `Model` to have with auto completion.

Giving us protection that only `ProvidesDynamicContext` implementations can be used here but still giving us `Model` auto-complete.

Reply:
Should be resolved.